app/controllers/service_provider_controller.rb

```diff
@@ -2,11 +2,27 @@ class ServiceProviderController < ApplicationController
   protect_from_forgery with: :null_session
 
   def update
-    if FeatureManagement.use_dashboard_service_providers?
-      ServiceProviderUpdater.new.run
-      SecureHeadersWhitelister.new.run
+    authorize do
+      if FeatureManagement.use_dashboard_service_providers?
+        ServiceProviderUpdater.new.run
+        SecureHeadersWhitelister.new.run
+      end
+
+      render json: { status: 'If the feature is enabled, service providers have been updated.' }
+    end
+  end
+
+  private
+
+  def authorize
+    if authorization_token == Figaro.env.dashboard_api_token
+      yield
+    else
+      render nothing: true, status: 401
     end
+  end
 
-    render json: { status: 'If the feature is enabled, service providers have been updated.' }
+  def authorization_token
+    request.headers['X-LOGIN-DASHBOARD-TOKEN']
```

Comment:
Just ran a curl while running the app locally:

```
identity-idp jy-shared-secret % curl -X POST --header "HTTP_X_LOGIN_DASHBOARD_TOKEN: test_token" localhost:3000/api/service_provider -i
HTTP/1.1 401 Unauthorized
Content-Type: text/plain; charset=utf-8
Cache-Control: no-cache
X-Request-Id: 4a66f131-f664-4aa3-b001-e13fe4ce83c8
X-Runtime: 0.016142
Content-Security-Policy: default-src 'self'; base-uri 'self'; block-all-mixed-content; child-src 'self'; connect-src 'self'; font-src 'self' data:; form-action 'self'; img-src 'self' data: *.google-analytics.com; media-src 'self'; object-src 'none'; script-src 'self' 'unsafe-eval' 'unsafe-inline'; style-src 'self' 'unsafe-inline'
X-Content-Type-Options: nosniff
X-Download-Options: noopen
X-Frame-Options: DENY
X-Permitted-Cross-Domain-Policies: none
X-XSS-Protection: 1; mode=block
Transfer-Encoding: chunked
Connection: close
Server: thin 1.5.1 codename Straight Razor

identity-idp jy-shared-secret % curl -X POST --header "X-LOGIN-DASHBOARD-TOKEN: test_token" localhost:3000/api/service_provider
{"status":"If the feature is enabled, service providers have been updated."}%
```

To me, this output suggests that the Rails app reads the header values as-is and does not expect underscores